cmd/api/v1/audit.go

```diff
@@ -0,0 +1,180 @@
+	"github.com/metal-stack/metal-lib/pkg/genericcli/printers"
+	"github.com/metal-stack/metal-lib/pkg/pointer"
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+	"google.golang.org/protobuf/types/known/timestamppb"
+)
+
+type audit struct {
+	c *config.Config
+}
+
+func newAuditCmd(c *config.Config) *cobra.Command {
+	a := &audit{
+		c: c,
+	}
+
+	cmdsConfig := &genericcli.CmdsConfig[*apiv1.AuditServiceGetRequest, *apiv1.AuditServiceGetRequest, *apiv1.AuditTrace]{
+		BinaryName:      config.BinaryName,
+		GenericCLI:      genericcli.NewGenericCLI[*apiv1.AuditServiceGetRequest, *apiv1.AuditServiceGetRequest, *apiv1.AuditTrace](a).WithFS(c.Fs),
+		Singular:        "audit trace",
+		Plural:          "audit traces",
+		Description:     "show audit traces of the api-server",
+		Sorter:          sorters.AuditSorter(),
+		OnlyCmds:        genericcli.OnlyCmds(genericcli.ListCmd, genericcli.DescribeCmd),
+		DescribePrinter: func() printers.Printer { return c.DescribePrinter },
+		ListPrinter:     func() printers.Printer { return c.ListPrinter },
+		ListCmdMutateFn: func(cmd *cobra.Command) {
+
+			cmd.Flags().String("request-id", "", "request id of the audit trace.")
+
+			cmd.Flags().String("from", "1h", "start of range of the audit traces. e.g. 1h, 10m, 2006-01-02 15:04:05")
+			cmd.Flags().String("to", "", "end of range of the audit traces. e.g. 1h, 10m, 2006-01-02 15:04:05")
+
+			cmd.Flags().String("user", "", "user of the audit trace.")
+			cmd.Flags().String("tenant", "", "tenant of the audit trace.")
+
+			cmd.Flags().String("project", "", "project id of the audit trace")
+
+			cmd.Flags().String("method", "", "api method of the audit trace.")
+			cmd.Flags().Int32("result-code", 0, "HTTP status code of the audit trace.")
+			cmd.Flags().String("source-ip", "", "source-ip of the audit trace.")
+
+			cmd.Flags().String("body", "", "filters audit trace body payloads for the giben text.")
+			cmd.Flags().String("error", "", "error of the audit trace.")
+
+			cmd.Flags().Int64("limit", 100, "limit the number of audit traces.")
+
+			genericcli.Must(cmd.RegisterFlagCompletionFunc("project", c.Completion.ProjectListCompletion))
+		},
+		DescribeCmdMutateFn: func(cmd *cobra.Command) {
+			cmd.Flags().Bool("prettify-body", false, "attempts to interpret the body as json and prettifies it.")
+		},
+	}
+
+	return genericcli.NewCmds(cmdsConfig)
+}
+
+func (a *audit) Get(id string) (*apiv1.AuditTrace, error) {
+	ctx, cancel := a.c.NewRequestContext()
+	defer cancel()
+
+	project := a.c.GetProject()
+	fmt.Printf("project: %s", project)
+	tenant, err := a.c.GetTenant()
+	if err != nil {
+		return nil, fmt.Errorf("tenant is required")
+	}
+
+	req := &apiv1.AuditServiceGetRequest{
+		Login: tenant,
+		Uuid:  id,
+	}
+
+	resp, err := a.c.Client.Apiv1().Audit().Get(ctx, connect.NewRequest(req))
+	if err != nil {
+		return nil, fmt.Errorf("failed to get audit trace: %w", err)
+	}
+
+	trace := resp.Msg.Audit
+
+	if viper.GetBool("prettify-body") {
```

Comment:
Reduce indention depth.

```suggestion
	if viper.GetBool("prettify-body") && trace.Body != nil {
```